src/boot/__tests__/SerializeEscaped-test.js

```diff
@@ -0,0 +1,148 @@
+    });
+
+    test('unescape leaves input untouched', () => {
+      const unescaped = SerializeEscaped.unescape(input);
+      expect(unescaped).toEqual(input);
+    });
+  });
+
+  describe('suffixed with garbage', () => {
+    const input = ESCAPE_CHARACTER + SERIALIZED_TYPE_FIELD_NAME + GARBAGE;
+    test('in escape, breaks match of SERIALIZED_TYPE_FIELD_NAME', () => {
+      const escaped = SerializeEscaped.escape(input);
+      expect(escaped).toEqual(input);
+    });
+
+    test('in unescape, breaks match of SERIALIZED_TYPE_FIELD_NAME', () => {
+      const unescaped = SerializeEscaped.unescape(input);
+      expect(unescaped).toEqual(input);
+    });
+  });
+
+  describe('garbage before ESCAPE_CHARACTER', () => {
+    const input = GARBAGE + ESCAPE_CHARACTER + SERIALIZED_TYPE_FIELD_NAME;
+    test('in escape, breaks match of SERIALIZED_TYPE_FIELD_NAME', () => {
+      const escaped = SerializeEscaped.escape(input);
+      expect(escaped).toEqual(input);
+    });
+
+    test('in unescape, breaks match of SERIALIZED_TYPE_FIELD_NAME', () => {
+      const unescaped = SerializeEscaped.unescape(input);
+      expect(unescaped).toEqual(input);
+    });
+  });
+
+  describe('garbage between ESCAPE_CHARACTER and SERIALIZED_TYPE_FIELD_NAME', () => {
+    const input = GARBAGE + ESCAPE_CHARACTER + SERIALIZED_TYPE_FIELD_NAME;
+    test('in escape, breaks match of SERIALIZED_TYPE_FIELD_NAME', () => {
+      const escaped = SerializeEscaped.escape(input);
+      expect(escaped).toEqual(input);
+    });
+
+    test('in unescape, breaks match of SERIALIZED_TYPE_FIELD_NAME', () => {
+      const unescaped = SerializeEscaped.unescape(input);
+      expect(unescaped).toEqual(input);
+    });
+  });
+});
+
+const escapeNTimes = (input: string, n: number): string => {
+  let output = input;
+  for (let i = 0; i < n; i++) {
+    output = SerializeEscaped.escape(output);
+  }
+  return output;
+};
+
+// Starts at a double-escape, so we don't duplicate the above test
+// 'single escape/unescape of SERIALIZED_TYPE_FIELD_NAME'.
+describe('double-, triple-, etc., escapes of SERIALIZED_TYPE_FIELD_NAME work, and repeated unescapes', () => {
+  // This can be relatively small. Likely outcomes are zero errors or
+  // ~MAX_ESCAPE_DEPTH errors; the latter can be annoying.
+  const MAX_ESCAPE_DEPTH = 10;
+
+  for (let n = 2; n <= MAX_ESCAPE_DEPTH; n++) {
+    const escapedNTimes = escapeNTimes(SERIALIZED_TYPE_FIELD_NAME, n);
+    test(`SERIALIZED_TYPE_FIELD_NAME escapes correctly ${n} time(s)`, () => {
+      expect(escapedNTimes).toEqual(ESCAPE_CHARACTER.repeat(n) + SERIALIZED_TYPE_FIELD_NAME);
+    });
+
+    const unescaped = SerializeEscaped.unescape(escapedNTimes);
+    test(`SERIALIZED_TYPE_FIELD_NAME, after ${n} escapes, unescapes correctly`, () => {
+      expect(unescaped).toEqual(escapeNTimes(SERIALIZED_TYPE_FIELD_NAME, n - 1));
+    });
+  }
+});
+
+describe('escape and unescape are inverses of each other', () => {
+  for (let i = 0; i < 100; i++) {
```

Comment:
I'm not sure why I chose 100 here. We can't (or *I* can't!) say with what probability this will flake, which was a prerequisite for using a randomized algorithm in `src/utils/__tests__/backoffMachine-test.js`, under 'timeouts are random from zero to 100ms, 200ms, 400ms, 800ms...'.